packages/react/src/components/SignInButton/SignInButton.tsx

```diff
@@ -28,8 +31,12 @@ import SignIn from '../SignIn/SignIn';
  * @param {ReactElement} props.customComponent - Optional custom component to be rendered.
  * @returns {ReactElement} Rendered SignInButton component.
  */
-const SignInButton = ({customComponent}: {customComponent?: ReactElement}): ReactElement => {
-  const [modalVisible, setModalVisible] = useState(false);
+const SignInButton = (props: SignInButtonProps): ReactElement => {
+  const {customComponent, showFooter = false, showLogo = false, showSignUp = false} = props;
+
+  const [modalVisible, setModalVisible] = useState<boolean>(false);
+
+  const authContext: AuthContext | undefined = useContext(AsgardeoContext);
```

Comment:
Instead of directly referring context, can't we use a custom hook to get these values? it looks more readable and elegant